### PR TITLE
SchemaView initialization from a PosixPath

### PR DESCRIPTION
**Pull request by joeflack4**
Updates
```
- Update: UX: SchemaView: Creating a new instance now accepts a PosixPath rather than just a string representation of a file path.
```

### CODE REVIEW

File: linkml_runtime/utils/schemaview.py [hunk 1]
```diff
@@ -111,9 +112,9 @@ class SchemaView(object):
     modifications: int = 0
     uuid: str = None
 
-    def __init__(self, schema: Union[str, SchemaDefinition],
+    def __init__(self, schema: Union[str, PosixPath, SchemaDefinition],
                  importmap: Optional[Mapping[str, str]] = None, merge_imports: bool = False):
-        if isinstance(schema, str):
+        if isinstance(schema, str) or isinstance(schema, PosixPath):
```
Comment (joeflack4) [inline linkml_runtime/utils/schemaview.py:117]: I had 3 different proposed ways to do this in [this issue](https://github.com/linkml/linkml/issues/1056), but I felt like this way was even simpler than those.

**Review by joeflack4 — COMMENTED** [review]
Addresses:
- https://github.com/linkml/linkml/issues/1056

**Review by sierra-moxon — APPROVED** [review]

**Comment by sierra-moxon** [comment]
@joeflack4 - looks great!  if possible, could you add or edit a test to take advantage of the new support for PosixPath?

**Review by cmungall — CHANGES_REQUESTED** [review]
Thanks!

I suggest generalizing this to work with other paths - see https://docs.python.org/3/library/pathlib.html - unless there is a reason to restrict?

**Comment by joeflack4** [comment]
@cmungall I was going to look at these requests this weekend. Thanks for handling! I see as well the need to convert it to a string as well within `__init__`, thanks for handling that.

**Pull request merged**